src/Authentication/Authentication/custom/common/Permissions.ps1

```diff
@@ -2,12 +2,15 @@
 #  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
 # ------------------------------------------------------------------------------
 
-$permissionsMsGraphServicePrincipal = $null
+
+$permissionsPulledFromMgGraphRequest = $null
 
 function Permissions_GetPermissionsData {
 
+    $permissionsMsGraphServicePrincipal = $null
+
     # 2. Making a REST request to MS Graph
-    if ($null -eq $script:permissionsMsGraphServicePrincipal){
+    if ($null -eq $permissionsMsGraphServicePrincipal){
```

Comment:
Now is probably a good time to say that the `permissionsMSGraphServicePrincipal` variable should match the naming convention of the functions, i.e. can you rename this variable throughout to use an underscore as in `permissions_MsGraphServicePrincipal`? That gets across the fact that this variable is considered a (static) 'member' of the `Permissions` 'class'.

Reply:
It's still missing the `_` underscore character -- can you make that change?